ta4j-core/src/test/java/org/ta4j/core/TestUtils.java

```diff
@@ -22,96 +22,93 @@ this software and associated documentation files (the "Software"), to deal in
  */
 package org.ta4j.core;
 
-import org.ta4j.core.num.Num;
-
 import java.math.BigDecimal;
-import java.util.List;
+
+import org.ta4j.core.num.Num;
 
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertNotEquals;
 
 /**
- * Utility class for {@code Decimal} tests.
+ * Utility class for {@code Num} tests.
  */
 public class TestUtils {
 
     /** Offset for double equality checking */
     public static final double GENERAL_OFFSET = 0.0001;
 
     /**
-     * Verifies that the actual {@code Decimal} value is equal to the given {@code String} representation.
+     * Verifies that the actual {@code Num} value is equal to the expected
+     * {@code String} representation within GENERAL_OFFSET.
      *
-     * @param actual the actual {@code Decimal} value
-     * @param expected the given {@code String} representation to compare the actual value to
-     * @throws AssertionError if the actual value is not equal to the given {@code String} representation
+     * @param actual the actual {@code Num} value
+     * @param expected the given {@code String} representation to compare the
+     *            actual value to
+     * @throws AssertionError if the actual value is not equal to the given
+     *             {@code String} representation
      */
     public static void assertNumEquals(String expected, Num actual) {
-        assertEquals(expected, actual.numOf(new BigDecimal(expected)).toString());
+        assertEquals(Double.valueOf(expected), actual.doubleValue(), GENERAL_OFFSET);
```

Comment:
The expected value could have higher precision as `Double ` type provides. Should be
`assertEquals(actual.numOf(new BigDecimal(expected)).toString(), actual);`

Reply:
Okay, will correct.